utils/dir-watcher/DirWatcher.js

```diff
@@ -0,0 +1,62 @@
+import fs from 'fs';
+import Emitter from 'events';
+
+class DirWatcher {
+  constructor(props) {
+    this.path = props.path;
+    this.delay = props.delay;
+    this.emitter = new Emitter();
```

Comment:
You can just extend DirWatcher from Emitter and delete this field (so instead of _this.emitter.emit_ you will call _this.emit_)